src/cart/SConscript

```diff
@@ -22,6 +22,18 @@ SRC = ['crt_bulk.c', 'crt_context.c', 'crt_corpc.c',
        'crt_swim.c', 'crt_tree.c', 'crt_tree_flat.c', 'crt_tree_kary.c',
        'crt_tree_knomial.c', 'crt_hlc.c', 'crt_hlct.c']
 
+LIB_SRC = ['crt_utils.c']
+
+####################################
+# Create cart utility share objects
+####################################
+def build_utility_shared_obj(denv):
+    """Build cart utility library"""
+    Import('prereqs')
+    prereqs.require(denv, 'mercury', headers_only=True)
```

Comment:
We should clone the environment rather than do any pollution to global environment that is passed in.

Reply:
74ad5fe2d513edf36957b4c26c9b2c1b7531e579